udata/rdf.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals, print_function
+'''
+This module centralize udata-wide RDF helpers and configuration
+'''
+from flask import request, url_for
+
+from rdflib import Graph
+from rdflib.namespace import (
+    Namespace, NamespaceManager, DCTERMS, SKOS, FOAF, XSD, RDFS
+)
+from rdflib.util import SUFFIX_FORMAT_MAP, guess_format as raw_guess_format
+from rdflib_jsonld.context import Context
+
+# Extra Namespaces
+ADMS = Namespace('http://www.w3.org/ns/adms#')
+DCAT = Namespace('http://www.w3.org/ns/dcat#')
+HYDRA = Namespace('http://www.w3.org/ns/hydra/core#')
+SCHEMA = Namespace('http://schema.org/')
+SPDX = Namespace('http://spdx.org/rdf/terms#')
+VCARD = Namespace('http://www.w3.org/2006/vcard/ns#')
+DCT = DCTERMS  # More common usage
+
+namespace_manager = NamespaceManager(Graph())
+namespace_manager.bind('foaf', FOAF)
+namespace_manager.bind('dct', DCT)
+namespace_manager.bind('skos', SKOS)
+namespace_manager.bind('dcat', DCAT)
+namespace_manager.bind('foaf', FOAF)
+namespace_manager.bind('hydra', HYDRA)
+namespace_manager.bind('vcard', VCARD)
+namespace_manager.bind('rdfs', RDFS)
+namespace_manager.bind('xsd', XSD)
+
+# Support JSON-LD in format detection
+FORMAT_MAP = SUFFIX_FORMAT_MAP.copy()
+FORMAT_MAP['json'] = 'json-ld'
+FORMAT_MAP['jsonld'] = 'json-ld'
+FORMAT_MAP['xml'] = 'xml'
+
+# Map serialization formats to MIME types
+RDF_MIME_TYPES = {
+    'xml': 'application/rdf+xml',
+    'n3': 'text/n3',
+    'turtle': 'application/x-turtle',
+    'nt': 'application/n-triples',
+    'json-ld': 'application/ld+json',
+    'trig': 'application/trig',
+    # Available but no activated
+    # 'nquads': 'application/n-quads',
+    # 'trix': 'text/xml',
+}
+
+# Map accepted MIME types to known formats
+ACCEPTED_MIME_TYPES = {
+    'application/rdf+xml': 'xml',
+    'application/xml': 'xml',
+    'text/n3': 'n3',
+    'application/x-turtle': 'turtle',
+    'text/turtle': 'turtle',
+    'application/n-triples': 'nt',
+    'application/ld+json': 'json-ld',
+    'application/json': 'json-ld',
+    'application/trig': 'trig',
+    # Available but no activated
+    # 'application/n-quads': 'nquads',
+    # 'text/xml': 'trix',
+}
+
+# Map formats to default used extensions
+RDF_EXTENSIONS = {
```

Comment:
These are main RDF supported extensions, supporting them comes without extra programming cost and choice has been discussed and validated with @ColinMaudry.
Different format for different usage/tools.
We can't predict what usage or tools will be use to consume udata rdf (not so many tools support `JSON-LD`). In fact, I've been using different tools during that PR and not all of them support `json-ld`.
On the harvest part, we need to be able to harvest these formats: not all portals expose json-ld which is the younger format in the list (this is why this is the only one which requires an extra depdency to `rdflib`).
The DCAT spec is a RDF specification, not JSON-LD specific (`DCAT` vocabulary itself is not exposed as `JSON-LD` but only `xml` and `turtle`)